src/client-side-encryption/client_encryption.ts

```diff
@@ -576,6 +576,8 @@ export class ClientEncryption {
               ...field,
               keyId: await this.createDataKey(provider, {
                 masterKey,
+                // clone the timeoutContext
+                // in order to avoid sharing the same timeout for server selection and connection checkout across different operations
```

Comment:
```suggestion
                // clone the timeoutContext
                // in order to avoid sharing the same timeout for server selection and connection checkout across different concurrent operations
```